src/views/darkroom.c

```diff
@@ -438,27 +470,33 @@ void expose(
     dev->gui_synch = FALSE;
   }
 
-  if(dev->image_status == DT_DEV_PIXELPIPE_DIRTY
-     || dev->image_status == DT_DEV_PIXELPIPE_INVALID
-     || dev->pipe->input_timestamp < dev->preview_pipe->input_timestamp)
+  dt_print(DT_DEBUG_EXPOSE, "[darkroom expose] initial requests: %s%s%s%s\n",
```

Comment:
For all those new `dt_print` routines we are calling 4 routines : `_*_request(dev)`.

Maybe worth enclosing `dt_print` around a condition to enter it only if DT_DEBUG_EXPOSE is set?

Reply:
Done. (Sorry for the noise, was falling back to FORTH thinking, that would have been a simple compiler extension with a CREATE DOES> word.)